convert2rhel/actions/pre_ponr_changes/subscription.py

```diff
@@ -185,7 +185,22 @@ def run(self):
                     return
                 raise
 
-            logger.warning("No rhsm credentials given to subscribe the system. Did not perform the subscription step.")
+            if not subscription.is_sca_enabled() and not subscription.is_subscription_attached():
+                logger.warning(
+                    "The system is registered with an RHSM account that has Simple Content Access (SCA) disabled but no subscription is attached. Without enabled SCA or an attached subscription the system can't access RHEL repositories. We'll try to auto-attach a subscription."
+                )
+                try:
+                    subscription.auto_attach_subscription()
+                except subscription.SubscriptionAutoAttachmentError:
+                    self.set_result(
+                        level="ERROR",
+                        id="NO_ACCESS_TO_RHEL_REPOS",
+                        title="No access to RHEL repositories",
+                        description="The system can access RHEL repositories only with either Simple Content Access (SCA) enabled or with an attached subscription.",
+                        diagnosis="The system is registered with an RHSM account that has SCA disabled but no subscription is attached. Auto-attaching a subscription was not successful.",
+                        remediations="Either attach a subscription manually by running 'subscription-manager attach --pool <pool id>' prior to the conversion or enable Simple Content Access on your RHSM account (https://access.redhat.com/articles/simple-content-access).",
```

Comment:
We also have an option in convert2rhel to set the pool id for the conversion, wouldn't the case here to tell the user to use our `--pool` instead of telling them to run subscription-manager? I was not following the work here closely, but, if we will unregister the system at rollback, then the system won't be registered in the first place, right?

Reply:
This particular case is to error when the user registers their system beforehand using subscription-manager but with an account without SCA enabled and no subs attached. About them using `--pool`, both options are viable - @bocekm recommended using the subscription-manager route

Reply:
Sounds good to me.